### PR TITLE
automate libiouring handling in cmake

### PR DESCRIPTION
**Pull request by gnusi**

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -263,7 +263,7 @@ set(IResearch_core_headers
   ./types.hpp
 )
 
-if (USE_URING)
+if (${uring_FOUND})
```
Comment (MBkkt) [inline core/CMakeLists.txt:266]:
```suggestion
if (uring_FOUND)
```

File: cmake/FindUring.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  set_target_properties(uring::uring PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${URING_INCLUDE_DIR}"
+    IMPORTED_LINK_INTERFACE_LANGUAGES "C"
+    IMPORTED_LOCATION "${URING_LIBRARIES}")
+endif()
+
+message("URING_INCLUDE_DIR: " ${URING_INCLUDE_DIR})
+message("URING_LIBRARIES: " ${URING_LIBRARIES})
+
+mark_as_advanced(URING_INCLUDE_DIR URING_LIBRARIES)
```
Comment (MBkkt) [inline cmake/FindUring.cmake:24]: I don't understand what is it doing.
In some examples it before `if (...` in other after

File: cmake/FindUring.cmake [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# URING_LIBRARIES - List of libraries when using uring
+# uring_FOUND - True if uring found
+
+find_path(URING_INCLUDE_DIR liburing.h)
+find_library(URING_LIBRARIES liburing.a liburing)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(uring DEFAULT_MSG URING_LIBRARIES URING_INCLUDE_DIR)
+
+if(uring_FOUND AND NOT TARGET uring::uring)
```
Comment (MBkkt) [inline cmake/FindUring.cmake:13]:
```suggestion
if (uring_FOUND AND NOT TARGET uring::uring)
```

File: cmake/FindUring.cmake (hunk 2 above)
Comment (gnusi) [reply cmake/FindUring.cmake:24]: I copied it from another `Find<...>.cmake` :-)

**Pull request merged**